usr/src/uts/common/brand/lx/os/lx_brand.c

```diff
@@ -2363,8 +2363,27 @@ lx_elfexec(struct vnode *vp, struct execa *uap, struct uarg *args,
 	 * library will ask us for this data later, when it is ready to set
 	 * things up for the lx executable.
 	 */
-	edp.ed_phdr = (uphdr_vaddr == -1) ? voffset + ehdr.e_phoff :
-	    voffset + uphdr_vaddr;
+	if (uphdr_vaddr == -1) {
+		edp.ed_phdr = voffset + ehdr.e_phoff;
+	} else if (ehdr.e_type == ET_EXEC && interp == NULL) {
+		/*
+		 * For statically linked executable, use vaddr from Phdr.
+		 */
+		edp.ed_phdr = uphdr_vaddr;
+	} else {
+		/*
+		 * Dynamic executable with found or determined Phdr.
+		 *
+		 *  - voffset is either address where exec file is mapped
+		 *   (for ET_DYN Type) or is zero.
+		 *
+		 * According to ELF format for PT_PHDR:
+		 *  - vaddr is the virtual address at which the first byte of the segment
+		 *    resides inmemory.
```

Comment:
missing space?

Reply:
You are right. It should be "in memory".

BTW, "ELF Format" document  is here [here](http://www.skyfree.org/linux/references/ELF_Format.pdf)

And more information fro PHDR.

**PT_PHDR**  The array element, if present, specifies the location and size of the program header table
itself, both in the file and in the memory image of the program. This segment type may
not occur more than once in a file. Moreover, it may occur only if the program header
table is part of the memory image of the program. If it is present, it must precede any
loadable segment entry. See ‘‘Program Interpreter’’ below for further information.

Reply:
Yep, thanks! Waiting for update now:)

Reply:
@tsoome  Can we leave it as is?

Reply:
It kind of defeats the purpose of this whole process. Well, of course we can apply the spelling fix as a follow up.  Why cant the PR request be updated? I mean, it should be just simple git push?

Reply:
This is from cneira:master

That commit already pushed into omnios repository, so spelling fix could be more suitable, I guess. And I like this word "inmemory"))

Reply:
ok lets make the spelling fix a separate one then, I do not mind:)

Reply:
Actually I'll pull this in with the fix.